torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2564,33 +2564,60 @@ def hint(s):
-                inner_sources_and_tensors = [(AttrSource(source, attr), getattr(t, attr)) for attr in attrs]
-                if t.is_nested:
-                    # For NestedTensors we need to track BOTH symints on the outer
-                    # tensor and tensor because we'd like to guard on the ragged
-                    # size but the symint representing ragged size is not in terms
-                    # of the symints on the inner tensors.
-                    sources_and_tensors.extend(inner_sources_and_tensors)
+
+                if is_traceable_wrapper_subclass(t):
+                    # If our placeholder is a tensor subclass, then the "true" symints
+                    # come from the subclass's inner tensors.
+                    attrs, _ = t.__tensor_flatten__()
+
+                    inner_sources_and_tensors = [(AttrSource(source, attr), getattr(t, attr)) for attr in attrs]
+                    if t.is_nested:
+                        # For NestedTensors we need to track BOTH symints on the outer
+                        # tensor and tensor because we'd like to guard on the ragged
+                        # size but the symint representing ragged size is not in terms
+                        # of the symints on the inner tensors.
+                        sources_and_tensors.extend([(source, t)])
+                        sources_and_tensors.extend(inner_sources_and_tensors)
+                    else:
+                        # For other tensor subclasses, only track the symints from
+                        # the inner tensors
+                        sources_and_tensors.extend(inner_sources_and_tensors)
                 else:
-                    # For other tensor subclasses, only track the symints from
-                    # the inner tensors
-                    sources_and_tensors = inner_sources_and_tensors
+                    sources_and_tensors.extend([(source, t)])
+
+                if t.is_nested and t._is_view():
+                    # We need to track symints on the base as well for NestedTensor views
+                    # because NT views are metafied in a different way than normal tensors.
+                    #
+                    # Ordinarily, the base is metafied with static DimDynamic so that
+                    # symints are not created for the sizes/strides of the base. Instead,
+                    # the symintification happens during view replay! This is possible
+                    # because as_strided conveniently sets the sizes/strides/storage_offsets
+                    # of the resulting tensor to symint values.
+                    #
+                    # However, NT and any other subclass that does not support as_strided
+                    # must rely instead on view_func to do this replaying. In this case,
+                    # the base must be metafied with dynamic DimDynamic and thus have
+                    # symints allocated, which then need to be tracked.
```

Comment:
Alternatively, it might be better to just symintify AFTER replaying the view_func.